packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -395,15 +417,15 @@ abstract class ServerlessClusterNew extends ServerlessClusterBase {
       databaseName: props.defaultDatabaseName,
       dbClusterIdentifier: clusterIdentifier,
       dbClusterParameterGroupName: clusterParameterGroupConfig?.parameterGroupName,
-      dbSubnetGroupName: subnetGroup.subnetGroupName,
+      dbSubnetGroupName: subnetGroup?.subnetGroupName,
       deletionProtection: defaultDeletionProtection(props.deletionProtection, props.removalPolicy),
       engine: props.engine.engineType,
       engineVersion: props.engine.engineVersion?.fullVersion,
       engineMode: 'serverless',
       enableHttpEndpoint: Lazy.any({ produce: () => this.enableDataApi }),
       scalingConfiguration: props.scaling ? this.renderScalingConfiguration(props.scaling) : undefined,
       storageEncrypted: true,
-      vpcSecurityGroupIds: this.securityGroups.map(sg => sg.securityGroupId),
+      vpcSecurityGroupIds: this.securityGroups?.map(sg => sg.securityGroupId),
```

Comment:
The `?` is not needed anymore, right?

```suggestion
      vpcSecurityGroupIds: this.securityGroups.map(sg => sg.securityGroupId),
```